### DISCUSSION TITLE
US Developer Support

### DISCUSSION THREAD

**Discussion by adambrakhane** [Hyundai-Kia-Connect/hyundai_kia_connect_api#481] [General]
Hey, it seems from the issues that the US API support lags behind the other regions. I'm in the US and would be interested in helping out. What has your process been previously for reverse engineering the API? I can help pick it up from there and try to keep the module tested and up to date in the US (kia) if you can get me started.

**Comment by cdnninja** [comment]
Yes it does.  Needs some love for sure!  Each region has a file for it.  In the USAs case Hyundai and Kia are separate as they function very differently.  

Let me know if you need a link to the files.  I'm not at a pc otherwise would do so now.  
The location based api implementation normalizes the data to vehicle.py class. 

As for sniffing the data I haven't ever done the Kia usa sniffing.   Typically for other regions I have used Knox to emulate android.   Older version of android seem to work around certificate pinning.  Then I used Charles proxy to sniff the ssl traffic.  

It will take playing to see what works for the current app.  

Other item you can do is look in the sensor data attributes.  That's the raw api response for a cached call.  If it has data that the integration supports you can just map it. No sniffing required.

**Reply by cdnninja** [reply]
I am excited for the interest by the way! I don't own a Kia anymore so don't spend much time maintaining this integration anymore.

**Comment by adambrakhane** [comment]
Cool! It's been a few years since I played with MITM tools, so it might take a little bit to get back into it, but I'll give it a shot.

Something about the SSL implementation doesn't work for me so I haven't tried the module much yet, but I reproduced all the API calls it makes in postman to check it out. It looks like they have changed some, but not all of the endpoints since you wrote this.
- Login ✅ 
- Get vehicles ✅ 
- Get vehicle state ✅ 
- Lock doors: 500 🚫 
- Unlock doors: 500 🚫
- Climate control: 500 🚫

_I haven't tested other commands like start/stop charge, charge scheduling, send map location, ..._



Have you found any of these endpoints to be car specific?

**Reply by cdnninja** [reply]
Some are generation specific in other regions.  Newer cars sometimes change things.

**Comment by cdnninja** [comment]
For the SSL topic I know for europe it helps to use android 6 non rooted with Charles to get past it.  Knox player as well.
